### PR TITLE
[SPARK-26716][SPARK-26765][FOLLOWUP][SQL] Clean up schema validation methods and override toString method in Avro

### PR DESCRIPTION
**Pull request by gengliangwang**
## What changes were proposed in this pull request?

In #23639, the API `supportDataType` is refactored. We should also remove the method `verifyWriteSchema` and `verifyReadSchema` in `DataSourceUtils`.

Since the error message use `FileFormat.toString` to specify the data source naming,  this PR also overriding the `toString` method in `AvroFileFormat`.

## How was this patch tested?

Unit test.

### CODE REVIEW

**Comment by gengliangwang** [comment]
@cloud-fan @dongjoon-hyun

File: external/avro/src/main/scala/org/apache/spark/sql/avro/AvroFileFormat.scala [hunk 1]
```diff
@@ -124,6 +124,8 @@ private[avro] class AvroFileFormat extends FileFormat
 
   override def shortName(): String = "avro"
 
+  override def toString(): String = "AVRO"
```
Comment (HyukjinKwon) [inline external/avro/src/main/scala/org/apache/spark/sql/avro/AvroFileFormat.scala]: I think avro is Avro like Parquet :)

**Review by HyukjinKwon — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**